ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/policyGen/policyBuilderOverlay_test.go

```diff
@@ -0,0 +1,327 @@
+package policyGen
+
+import (
+	utils "github.com/openshift-kni/cnf-features-deploy/ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/utils"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/yaml.v3"
+)
+
+// Take the string input and build the policies by calling PolicyBuilder.Build
+// and using the generic source-cr test data. Return the output of Build()
+func buildTest(t *testing.T, input string) (map[string]interface{}, error) {
+	// Read in the test PGT
+	pgt := utils.PolicyGenTemplate{}
+	_ = yaml.Unmarshal([]byte(input), &pgt)
```

Comment:
For debugging purposes, it's useful to just set `err := ...` and then `assert.NoError(t, err)` -> If you ever accidentally pass in bad input, it makes debugging so much easier.

Reply:
Good point. This is a backport from the master branch. I'm happy to update it there but would rather not make changes here in order to make it easier to track the backports.